src/Common/Thermodynamics/relations.jl

```diff
@@ -1105,9 +1105,25 @@ function saturation_adjustment(
             maxiter,
         )
         if !sol.converged
+            @print("-----------------------------------------\n")
             @print("maxiter reached in saturation_adjustment:\n")
+            @print(
+                "    e_int=",
+                e_int,
+                ", ρ=",
+                ρ,
+                ", q_tot=",
+                q_tot,
+                ", T = ",
+                sol.root,
+                ", maxiter=",
+                maxiter,
+                ", tol=",
+                tol.tol,
+                "\n"
+            )
             if error_on_non_convergence()
-                error("Exiting to avoid excessively large output logs")
+                throw(ArgumentError("Failed to converge with printed set of inputs."))
```

Comment:
Did you check that this worked on the GPU? `error` might be a safer choice, but if this works :+1:

Reply:
Not yet. Can I still test `error`s with `@test_throws`?

Reply:
Yeah you should get an `ErrorException`